fiona/collection.py

```diff
@@ -147,7 +148,10 @@ def __init__(self, path, mode='r', driver=None, schema=None, crs=None,
                 if 'init' in crs or 'proj' in crs or 'epsg' in crs.lower():
                     self._crs = crs
                 else:
-                    raise CRSError("crs lacks init or proj parameter")
+                    crs_wkt = crs_to_wkt(crs)
+                    if not crs_wkt:
+                        raise CRSError("Invalid input for CRS: {}".format(crs))
```

Comment:
```suggestion
                        raise CRSError("Invalid input for crs: {}".format(crs))
```

(the keyword argument is lower case?)

Reply:
Oh ha, I guess it depends on what you are referring to. I was thinking that the input was invalid for creating a CRS, but I see that can be confused with the kwarg. I was actually thinking of moving this logic inside the `crs_to_wkt` function. Thoughts?

Reply:
Updated

Reply:
> I was thinking that the input was invalid for creating a CRS, but I see that can be confused with the kwarg.

The reason that I was confused was because my first reaction "but fiona has no CRS (*capital letters indicating an object*) class". But of course, it still represents the notion of a CRS, so my confusion was only of short duration, so I think your original one is also fine.